### ISSUE TITLE
Hammer Editor in Linux

### ISSUE THREAD

**Issue by Antiaris74** [ValveSoftware/source-sdk-2013#141]
There's no Hammer Editor, Model Viewer and Face Poser in Source SDK Base 2013 for Linux.

**Comment by aaronfranke** [comment]
Why was this closed without a statement from Valve? A definitive yes/no would be preferable, but even "we'll look into it" is better than silence.

**Comment by petrikov** [comment]
It kind of works under wine but not without hiccups and some model/texture problems - which i wouldn't suggest as an eligible development environment.

**Comment by BerntA** [comment]
Why not create these tools yourselves? Most of the necessary code to 'glue' this together is available in the SDK2013 code?

**Comment by TheLazyCWriter** [comment]
VALVe please, don't do this whole silence thing again, just say "maybe" or just plainly "no" instead of keeping us in the dark here.
I'd love to get into mapping, but I'd hate to have to restart my PC all the time just to open one application. We've seen the success of the TF2 port to Linux, not to mention other games like CS:GO and L4D2, why not port the hammer editor?

**Comment by sirtomato999** [comment]
PLEASE VALVE

**Comment by JaxWilko** [comment]
With the release of proton, is there any update to this?

**Comment by sirtomato999** [comment]
if you make it cross platform mac will be able to use it to you know

**Comment by SourceSkyBoxer** [comment]
Grrr why do you not care to us? If we buy Half-Life 2 than we want create maps for Linux and Mac OS.

DO NOT USE ONLY WINDOWS! We are not Windows Users just We are Linux- and Mac-Users.

STOP IGNORING LONGER!

IF YOU DON'T RELEASE THAN YOU WILL LOOSE If We leave Valve Software because we are mad

PLEASE PROMISE US! 
I WANT WE HAVE HAMMER FOR LINUX AND MACOS

IF YOU NEVER RELEASE THAN WE IGNORE YOU BACK. WE RELEASE OWN BECAUSE YOU ALWAYS ARE LAZY COWBOYS, Valve Software!

Thank you for hurting If we are stupid to buy Half-Life and Mods and Source Engine.
WE ARE VERY DISAPPOINTED!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by KeironO** [comment]
Valve pls fix

**Comment by andrealmeid** [comment]
Since there is no native version, I wrote a guide on how to run it in Linux: https://andrealmeid.com/post/2020-05-28-csgo-hammer-linux/

### TL;DR

Run this in Steam console:

```
download_depot 730 732
```

Then in your terminal:

```
cd ~/.steam/steam && cp -r ubuntu12_32/steamapps/content/app_730/depot_732/* "steamapps/common/Counter-Strike Global Offensive"
```

Append `.vmf` at all `$files` without any extesion in expert options. The tool
will be ready to open and work. Beware leaks in the map. Save the map in the
default folder (`~/.steam/steam/steamapps/common/Counter-Strike Global
Offensive/sdk_content/maps/`). Open the Linux version of CS:GO, type in console
`map <map_name>` and have fun.

**Comment by MSafariyan** [comment]
It's 2021 valve.
release a Linux version of your hammer editor.

**Comment by Jan200101** [comment]
Since people still have not understood it:
Source 1 Hammer will **never** come to Linux or Mac
the codebase depends too much on Windows functionality and they do not care about rewriting
begging or being a complete asswipe is pointless
your options are:
- spin up a VM
- dualboot
- use wine [(comment)](https://github.com/ValveSoftware/source-sdk-2013/issues/141#issuecomment-635743520)
- make your own editor
- use the [sdk](https://github.com/ValveSoftware/source-sdk-2013) to implement your own map format

**Comment by MSafariyan** [comment]
> Since people still have not understood it:
> Source 1 Hammer will **never** come to Linux or Mac
> the codebase depends too much on Windows functionality and they do not care about rewriting
> begging or being a complete asswipe is pointless
> your options are:
> 
>     * spin up a VM
> 
>     * dualboot
> 
>     * use wine [(comment)](https://github.com/ValveSoftware/source-sdk-2013/issues/141#issuecomment-635743520)
> 
>     * make your own editor
> 
>     * use the [sdk](https://github.com/ValveSoftware/source-sdk-2013) to implement your own map format

Okey
at least I hope they do something for Linux in source 2.

**Comment by acoolstraw** [comment]
> The tools are Windows-only.

mlukni be cheren

**Comment by TheAwesome98-Real** [comment]
> Since people still have not understood it:
> Source 1 Hammer will **never** come to Linux or Mac
> the codebase depends too much on Windows functionality and they do not care about rewriting
> begging or being a complete asswipe is pointless
> your options are:
> 
>     * spin up a VM
> 
>     * dualboot
> 
>     * use wine [(comment)](https://github.com/ValveSoftware/source-sdk-2013/issues/141#issuecomment-635743520)
> 
>     * make your own editor
> 
>     * use the [sdk](https://github.com/ValveSoftware/source-sdk-2013) to implement your own map format

said the asswipe

anyway, back on topic, i'm writing a open-source hammer editor for linux and macOS that can also do src2, goldsrc, and possibly quake. check my repos for openhammer if you want to contribute and/or see the progress

**Comment by Jan200101** [comment]
> said the asswipe

as it stands that is simply a matter of facts and no reason to use ad hominem.
The way Hammer was written does not lend itself to porting easily and even when using a compatibility layer like wine many graphical and layout issues pop-up

> anyway, back on topic, i'm writing a open-source hammer editor for linux and macOS that can also do src2, goldsrc, and possibly quake. check my repos for openhammer if you want to contribute and/or see the progress

I do wish you good luck in your endevours
but your goal does seem rather amibtious considering this is something not even the most talented people in the community managed to do

**Comment by Yetoo1** [comment]
@Jan200101 You missed the part where MSafariyan called him an asswipe first.

**Comment by TheAwesome98-Real** [comment]
> this is something not even the most talented people in the community managed to do

look at J.A.C.K, it’s what I’m making but older and doesn’t support source or source 2

**Comment by ArturAiwasan** [comment]
@TheAwesome98-Real ждём...

**Comment by theangrygrunt** [comment]
@TheAwesome98-Real any progress on the tool?

**Comment by TheAwesome98-Real** [comment]
> @TheAwesome98-Real any progress on the tool?

not good so far, i'm rewriting the tool in rust because of memory safety. to create maps, what i do now is just install the windows version of the game somewhere, and run hammer using a newer version of wine.

**Comment by ArturAiwasan** [comment]
> i'm rewriting the tool in rust because of memory safety
> because of memory safety

ok (ne)

**Comment by TheAwesome98-Real** [comment]
what's wrong with that, i hate c++

**Comment by ArturAiwasan** [comment]
> what's wrong with that, i hate c++

Aspektas tiel ĝi, ke vi, simple, mensogas :eyes: 
Ĉu havas vi ajn ion, kio estas ligita kun nova VHE por Linukso? Pratipon ia aŭ ion simila. Per ĉi Rust.

**Comment by joshuachris2001** [comment]
why can't just the hammer interface be ported, if needed to can't the core backed just stay windows? for instance a cross platform interface, and on map compile/etc use a wine environment
=-=BUMP=-=

**Comment by acoolstraw** [comment]
why is this closed?

**Comment by theangrygrunt** [comment]
> why is this closed?

valve has no plans on porting sdk tools to linux due to stuff like Hammer using obscure windows binaries; best we can hope is that source 2's sdk (or at least Hammer in CS:2 will) will be crossplatform.
